File: ios/DataModels/PrimerExceptionRN.swift
struct PrimerExceptionRN: Encodable {
    let exceptionType: ExceptionTypeRN
    let description: String?
}

enum ExceptionTypeRN: String, Error, Encodable {
    case ParseJsonFailed
    case noIosViewController
    case invalidPrimerIntent
    case themeParsingFailed
    case settingsParsingFailed
    case settingsNotConfigured
    case clientTokenNotConfigured
    case flowParsingFailed
}

File: android/src/main/java/com/primerioreactnative/datamodels/PrimerExceptionRN.kt
package com.primerioreactnative.datamodels

import kotlinx.serialization.Serializable


@Serializable
data class PrimerExceptionRN(
  val exceptionType: ExceptionTypeRN,
  val description: String? = null,
)

@Serializable
enum class ExceptionTypeRN {
  ParseJsonFailed,
  InitFailed,
  CheckoutFlowFailed,
  TokenizationFailed,
}
